protocol/directory.go

```diff
@@ -39,12 +39,15 @@ func (d *ConiksDirectory) Update() {
 	}
 }
 
-func (d *ConiksDirectory) SetPolicies(epDeadline merkletree.TimeStamp, vrfKey vrf.PrivateKey) {
-	d.policies = merkletree.NewPolicies(epDeadline, vrfKey)
+func (d *ConiksDirectory) UpdatePolicies(epDeadline merkletree.TimeStamp, vrfKey vrf.PrivateKey) {
+	d.policies.Update(epDeadline, vrfKey)
 }
 
+// EpochDeadline returns the EpochDeadline of the current policies
+// If you want to query the EpochDeadline before the new policies
+// is updated, use LatestSTR().Policies.EpochDeadline
```

Comment:
But that's exactly what you want to use in `EpochUpdate`.

https://github.com/coniks-sys/coniks-go/blob/master/keyserver/server.go#L192

The timer needs to fire based on `LatestSTR`'s deadline.

Again, I really think you should revert everything here that isn't about removing the interface.  The new test is good to add, but orthogonal.